src/lib/site-server-process.ts

```diff
@@ -142,14 +143,26 @@ export default class SiteServerProcess {
 		}
 
 		return new Promise< void >( ( resolve, reject ) => {
+			let isResolved = false;
 			process.once( 'exit', ( code ) => {
-				if ( code !== 0 ) {
-					reject( new Error( `Site server process exited with code ${ code } upon stopping` ) );
-					return;
+				if ( ! isResolved ) {
+					isResolved = true;
```

Comment:
```suggestion
```
Nit, but given that we're only listening to the `exit` event once, do we really need to modify `isResolved` here?